src/main/scala/firrtl_interpreter/vcd/VCDConfig.scala

```diff
@@ -2,54 +2,102 @@
 
 package firrtl_interpreter.vcd
 
-import firrtl.ExecutionOptionsManager
+import firrtl.AnnotationSeq
+import firrtl.annotations.NoTargetAnnotation
+import firrtl.options.{HasScoptOptions, OptionsView, RegisteredLibrary}
+import scopt.OptionParser
 
 case class VCDConfig(
-                     vcdSourceName:       String = "",
-                     vcdTargetName:       String = "",
-                     startScope:          String = "",
-                     renameStartScope:    String = "",
-                     varPrefix:           String = "",
-                     newVarPrefix:        String = "")
-  extends firrtl.ComposableOptions {
-}
+  vcdSourceName:       String = "",
+  vcdTargetName:       String = "",
+  startScope:          String = "",
+  renameStartScope:    String = "",
+  varPrefix:           String = "",
+  newVarPrefix:        String = ""
+)
 
-trait HasVCDConfig {
-  self: ExecutionOptionsManager =>
+object VcdConfigViewer {
+  implicit object VcdConfigView extends OptionsView[VCDConfig] {
+    def view(options: AnnotationSeq): Option[VCDConfig] = {
+      val executionOptions = options.foldLeft(VCDConfig()) { (previousOptions, annotation) =>
+        annotation match {
+          case VcdSourceNameAnnotation(name)       => previousOptions.copy(vcdSourceName = name)
+          case VcdTargetNameAnnotation(name)       => previousOptions.copy(vcdTargetName = name)
+          case VcdStartScopeAnnotation(name)       => previousOptions.copy(startScope = name)
+          case VcdRenameStartScopeAnnotation(name) => previousOptions.copy(renameStartScope = name)
+          case VcdVarPrefixAnnotation(name)        => previousOptions.copy(varPrefix = name)
+          case _ => previousOptions
+        }
 
-  var vcdConfig = VCDConfig()
+      }
+      Some(executionOptions)
+    }
+  }
+}
 
-  parser.note("vcd")
+sealed trait VcdConfigOption extends HasScoptOptions
 
-  parser.opt[String]("start-scope")
-    .abbr("vss")
-    .foreach { x => vcdConfig = vcdConfig.copy(startScope = x)}
-    .text("starts saving information at specified scope")
+case class VcdSourceNameAnnotation(name: String = "") extends NoTargetAnnotation with VcdConfigOption {
+  def addOptions(p: OptionParser[AnnotationSeq]): Unit = p.opt[String]("vcd-start-scope")
```

Comment:
This option is a duplicate of the long option for `VcdStartScopeAnnotation`.

Reply:
fixed typo in VcdSourceNameAnnotation